qt-bugs@ issue : N64978
applied: no
author: Germain Garand <germain@ebooksfrance.org>

In QScrollView, wheel events are forwarded to (viewport|contents)WheelEvent, but
the horizontal/vertical status of the wheel event is not.

Index: src/widgets/qscrollview.cpp
===================================================================
RCS file: /home/kde/qt-copy/src/widgets/qscrollview.cpp,v
retrieving revision 1.65
diff -u -3 -p -r1.65 qscrollview.cpp
--- src/widgets/qscrollview.cpp	29 Apr 2004 22:31:28 -0000	1.65
+++ src/widgets/qscrollview.cpp	25 Jan 2005 12:07:04 -0000
@@ -1180,7 +1180,7 @@ void  QScrollView::mouseMoveEvent( QMous
 void QScrollView::wheelEvent( QWheelEvent *e )
 {
     QWheelEvent ce( viewport()->mapFromGlobal( e->globalPos() ),
-                    e->globalPos(), e->delta(), e->state());
+                    e->globalPos(), e->delta(), e->state(), e->orientation() );
     viewportWheelEvent(&ce);
     if ( !ce.isAccepted() ) {
 	if ( e->orientation() == Horizontal && horizontalScrollBar() )
@@ -1863,7 +1863,7 @@ void QScrollView::viewportWheelEvent( QW
        the event itself.
     */
     QWheelEvent ce( viewportToContents(e->pos()),
-        e->globalPos(), e->delta(), e->state());
+        e->globalPos(), e->delta(), e->state(), e->orientation());
     contentsWheelEvent(&ce);
     if ( ce.isAccepted() )
         e->accept();
